test-data/unit/check-newsemanal.test

```diff
@@ -1970,7 +1970,7 @@ class G(Generic[T]): pass
 
 t: Tuple[G[B], G[C]] # E: Type argument "__main__.B" of "G" must be a subtype of "__main__.A" \
                      # E: Type argument "__main__.C" of "G" must be a subtype of "__main__.A"
-reveal_type(t.__iter__) # N: Revealed type is 'def () -> typing.Iterator[__main__.G*[__main__.B]]'
+reveal_type(t.__iter__) # N: Revealed type is 'def () -> typing.Iterator[builtins.object*]'
```

Comment:
Previously mypy ignored the typevar not being a subtype of the bound.
Now `object` is inferred instead. This shouldn't matter, as this can only happen with as the result of an error.